charts/kibana/templates/_helpers.tpl

```diff
@@ -64,6 +64,18 @@ imagePullSecrets:
 {{- end -}}
 {{- end -}}
 
-{{ define "kibana.IndexPattern" -}}
-{{- if .Values.global.loggingSidecar.enabled }}vector{{- else }}fluentd{{- end -}}
-{{- end }}
+{{- define "kibana.IndexPattern" -}}
+{{- if .Values.global.loggingSidecar.enabled }}
+{{- if .Values.global.logging.indexNamePrefix -}}
+{{- .Values.global.logging.indexNamePrefix -}}
+{{- else -}}
+vector
+{{- end -}}
+{{- else -}}
+{{- if .Values.global.logging.indexNamePrefix -}}
+{{- .Values.global.logging.indexNamePrefix -}}
+{{- else -}}
+fluentd
+{{- end -}}
+{{- end -}}
+{{- end -}}
```

Comment:
I think this can be simplified with the same code suggestion as external-es-proxy.